cpp/arcticdb/column_store/column_data.hpp

```diff
@@ -215,30 +216,46 @@ struct ColumnData {
         };
     }
 
+    /// @brief Get non-owning pointer to the shapes array for the column
+    [[nodiscard]] const Buffer* shapes() const noexcept;
+
   private:
     template<typename TDT>
     TypedBlockData<TDT> next_typed_block(MemBlock* block) {
         size_t num_elements = 0;
         const shape_t *shape_ptr = nullptr;
 
         constexpr auto dim = TDT::DimensionTag::value;
-        if constexpr (dim == Dimension::Dim0){
+        if constexpr (dim == Dimension::Dim0) {
             num_elements = block->bytes() / get_type_size(type_.data_type());
-        }  else {
+        } else {
             if (shapes_->empty()) {
                 num_elements = block->bytes() / get_type_size(type_.data_type());
             } else {
                 shape_ptr = shapes_->ptr_cast<shape_t>(shape_pos_, sizeof(shape_t));
                 size_t size = 0;
                 constexpr auto raw_type_sz = sizeof(typename TDT::DataTypeTag::raw_type);
-                while (size < block->bytes()) {
-                    if constexpr (dim == Dimension::Dim1)
+                // Blocks can contain empty tensors (empty arrays/matrices). They don't hold any data, however, they
+                // have assigned shapes. In case of an empty tensor the corresponding entries in the shapes array must
+                // be 0. The rationale is that the shapes array describes how many elements are there in a tensor. This
+                // way we can distinguish between [] (empty array) and None (no array at all). We assume that no block,
+                // besides the first, can start with empty tensors. This means that a block is exhausted when both are
+                // conditions are satisfied:
+                // i) The processed size becomes equal to the block size
+                // ii) All zero-sized shapes are parsed (i.e. the shape of the current tensor is not 0)
+                while (is_current_tensor_empty() || size < block->bytes()) {
```

Comment:
maybe current_tensor_is_empty, as it reads more like a normal sentence structure